### PR TITLE
test(batch-selection): Playwright refactor 

### PR DESCRIPTION
**Pull request by ZhuoyuJin**
### Proposed behaviour

The Batch-Selection component should have in-browser tests using the Playwright framework

### Current behaviour

Batch-Selection currently has Cypress tests instead of Playwright tests

### Checklist

- [X] Commits follow our style guide
- [X] Related issues linked in commit messages if required
- [X] Screenshots are included in the PR if useful
- [X] All themes are supported if required
- [X] Unit tests added or updated if required
- [X] Cypress automation tests added or updated if required
- [X] Playwright automation tests added or updated if required
- [X] Storybook added or updated if required
- [X] Translations added or updated (including creating or amending translation keys table in storybook) if required
- [X] Typescript `d.ts` file added or updated if required
- [X] Related docs have been updated if required

#### QA

- [ ] Tested in CodeSandbox/storybook
- [ ] Add new Cypress test coverage if required
- [ ] Carbon implementation matches Design System/designs
- [ ] UI Tests GitHub check reviewed if required

### Testing instructions

The following CodeSandbox is an example of the broken behaviour.
You can see the new behaviour by looking at the version in the comment by `codesandbox[bot]`.

### CODE REVIEW

File: src/components/batch-selection/batch-selection.pw.tsx [hunk 1]
```diff
@@ -0,0 +1,191 @@
+import React from "react";
+import { test, expect } from "@playwright/experimental-ct-react17";
+import { BatchSelectionProps } from "components/batch-selection";
```
Comment (stephenogorman) [inline src/components/batch-selection/batch-selection.pw.tsx:3]: Could you add a proper path to this import?

File: src/components/batch-selection/batch-selection.pw.tsx [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+    await expect(batchSelectionComponent(page).getAttribute("hidden"))
+      .toBeTruthy;
+
+    await expect(batchSelectionComponent(page)).toBeVisible();
+  });
+
+  test("should check disabled BatchSelection", async ({ mount, page }) => {
+    await mount(<BatchSelectionComponent disabled />);
+
+    await expect(batchSelectionComponent(page).getAttribute("disabled"))
```
Comment (DipperTheDan) [inline src/components/batch-selection/batch-selection.pw.tsx:50]:
```suggestion
    await expect(batchSelectionComponent(page)).toBeDisabled();
```

Comment (DipperTheDan) [reply src/components/batch-selection/batch-selection.pw.tsx:50]: You wouldn't need the `.toBeTruthy;` on the next line either

File: src/components/batch-selection/batch-selection.pw.tsx [hunk 3]
```diff
@@ -0,0 +1,191 @@
...
+    });
+  });
+
+  test("should check hidden BatchSelection", async ({ mount, page }) => {
+    await mount(<BatchSelectionComponent hidden />);
+
+    await expect(batchSelectionComponent(page).getAttribute("hidden"))
+      .toBeTruthy;
+
+    await expect(batchSelectionComponent(page)).toBeVisible();
```
Comment (DipperTheDan) [inline src/components/batch-selection/batch-selection.pw.tsx:44]: **issue:** I wouldn't expect `.toBeVisible()`to pass here considering that the prop is called `hidden`. I'd expect the correct assertion to be `.not.toBeVisible()`.

File: src/components/batch-selection/batch-selection.pw.tsx [hunk 4]
```diff
@@ -0,0 +1,191 @@
...
+  );
+});
+
+test.describe("check BatchSelection buttons are focused", () => {
+  (["first", "second", "third"] as const).forEach((index) => {
+    test(`should check BatchSelection ${index} button has expected styling when focusRedesignOptOut is true`, async ({
+      mount,
+      page,
+    }) => {
+      await mount(<BatchSelectionComponent selectedCount={1} />, {
```
Comment (DipperTheDan) [inline src/components/batch-selection/batch-selection.pw.tsx:91]:
```suggestion
      await mount<HooksConfig>(<BatchSelectionComponent selectedCount={1} />, {
```

Comment (DipperTheDan) [reply src/components/batch-selection/batch-selection.pw.tsx:91]: You'll need to import `HooksConfig` too but VS Code should be able to sort that import out for you 😄

File: src/components/batch-selection/batch-selection.pw.tsx [hunk 5]
```diff
@@ -0,0 +1,191 @@
...
+      mount,
+      page,
+    }) => {
+      await mount(<BatchSelectionComponent selectedCount={1} />);
+
+      const elementLocator = batchSelectionButtonsByPosition(
+        page,
+        positionOfElement(index)
+      );
+      const element = await elementLocator;
```
Comment (DipperTheDan) [inline src/components/batch-selection/batch-selection.pw.tsx:119]: **suggestion:** Not sure you need the await here.

**Review by DipperTheDan — APPROVED** [review]

**Pull request merged**